### PR TITLE
[COMGR] Use OpenCL 2.0. [HIPRTC] Provide min/max limits for int. Fix build errors related to min/max limits for BF16.

### PR DESCRIPTION
**Pull request by atamazov**
This is a companion PR for #2691 that resolves the remaining issues with mainline HIP runtime. Details:

- Fixed runtime issues:
  - [COMGR] Use OpenCL 2.0 for all kernels. Allows non-uniform grids for OCL kernels with RC HIP runtime.
- Fixed errors during kernel builds:
  - [HIPRTC] Provide min/max limits for int
  - Removed `constexpr` from numeric_limits<hip_bfloat16>::min()/max() as BF16 ctor provided by HIP can't be used in const expressions.

## :cyclone: Performance testing results

Preconditions:
- Navi21 (gfx1030/36) with ROCm 6.0 release docker
- BUILD_DEV=On, Release build
- 93 known FP32 convolution configs
- 93 known FP16 convolution configs
- NORMAL Find mode, only OpenCL convolutions enabled

Bottom line: :green_circle:  No performance or correctness regressions. Detailed logs & csv files are available upon request.

----
[Attribution] @junliume @JehandadKhan
- https://github.com/ROCm/MIOpen/labels/urgency_blocker
- Proposed reviewers: @JehandadKhan, @junliume

### CODE REVIEW

**Comment by atamazov** [comment]
## :dart: Correctness testing results

Preconditions:
- Navi21 (gfx1030/36) with _mainline HIP (6.1.24024)_
- BUILD_DEV=On, Release build, MIOPEN_TEST_ALL=On

`make check` with FLOAT, HALF, BFLOAT16, INT8 -- all passed 🟢

**Review by junliume — APPROVED** [review]
